### PR TITLE
Gui/feature/timeline

### LINKED ISSUE #110: Add a Timeline to the GUI (Library 3)
## User story
1. As a user
2. I want / need a good GUI 
3. So that I can effectively and efficiently analyze the differences. 

## Acceptance criteria
* There should be a timeline that shows the percentage and the total minutes as shown in the ui prototype. 
* There should be a indicator that shows at which position the video is in the timeline
* The indicator should show the percentage of the current progress
* The timeline of the GUI should look like as illustrated below 
* By clicking on a timestep in the timeline, one can directly jump to the screen at that time.
* Show absolute frames above the timeline beginning and end

Image

**Hint: In the scope of this ticket is not the preview of the video in the timeline!**

## Definition of done (DoD)
* Create feature branch
* Implement the functionality
* Opened pull request and assigned reviewer
* If necessary: required changes are implemented
* Delete feature branch
* If necessary: Component Test has been written
* Code review has been completed and code has been merged

## DoD general criteria
* Feature has been fully implemented
* Feature has been merged into the mainline
* All acceptance criteria were met
* Product owner approved features
* All tests are passing
* Developers agreed to release

### PR DESCRIPTION
**Pull request by AlperK61**
closes #110

### CODE REVIEW

**Comment by akriese** [comment]
Can you rebase this onto the current main and resolve conflicts? Then the CI will be able to run :)

**Comment by akriese** [comment]
Tried it out and it looks pretty nice to me! Good work :)
One thing that I would change is the red stroke jumping to the actual percentage of the closest frame.
So, if there are 10 images, I click at 54%, it should directly go to 50%. What do you think?

File: GUI/src/main/kotlin/Timeline.kt [hunk 1]
```diff
@@ -54,6 +54,7 @@ class Timeline {
                 )
             }
 //            #### timeline box ####
+            var helper = Offset.Zero
```
Comment (akriese) [inline GUI/src/main/kotlin/Timeline.kt:57]: Could you rename this variable to a more descriptive one? I think, it is used to store the actual mouse position, so maybe `mousePosition`?

File: GUI/src/main/kotlin/Timeline.kt [hunk 2]
```diff
@@ -0,0 +1,152 @@
+import androidx.compose.foundation.Canvas
```
Comment (akriese) [inline GUI/src/main/kotlin/Timeline.kt:1]: I think this file should be placed in the ui/components/ folder to stick with the current file tree scheme

**Review by akriese — CHANGES_REQUESTED** [review]

File: GUI/src/main/kotlin/ui/components/Timeline.kt [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.input.pointer.pointerInput
+import androidx.compose.ui.layout.layout
+import androidx.compose.ui.layout.onGloballyPositioned
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import frameNavigation.FrameNavigation
+
+class Timeline {
```
Comment (simonsasse) [inline GUI/src/main/kotlin/ui/components/Timeline.kt:22]: I think usually one does not wrap composeables into a class. I suggest removing the `class Timeline` and renaming the `fun timeline` to `fun Timeline`.

**Review by simonsasse — CHANGES_REQUESTED** [review]
Works fine and code looks good :) Only one thing, composable are usually not wrapped in a class.

**Review by simonsasse — APPROVED** [review]
LGTM :)

**Review by akriese — APPROVED** [review]

**Pull request merged**